### PR TITLE
Added the switch for the inclusion of PS energy

### PR DESCRIPTION
**Pull request by ram1123**
#### PR description:

EGamma regression can be trained with and without PS energy taking into consideration. So, we added a switch for this while performing the evaluation in the CMSSW.

***This PR is relevant ONLY for HLT supercluster regression, and that it does not effect the offline egamma regression.***

#### PR validation:

We run the code with the current version of update and it worked as expected.

### CODE REVIEW

File: RecoEcal/EgammaClusterAlgos/interface/SCEnergyCorrectorSemiParm.h [hunk 1]
```diff
@@ -146,6 +147,7 @@ template <edm::Transition esTransition>
 void SCEnergyCorrectorSemiParm::setTokens(const edm::ParameterSet& iConfig, edm::ConsumesCollector cc) {
   isHLT_ = iConfig.getParameter<bool>("isHLT");
   isPhaseII_ = iConfig.getParameter<bool>("isPhaseII");
+  regTrainedWithPS_ = iConfig.getParameter<bool>("regTrainedWithPS");
```
Comment (thomreis) [inline RecoEcal/EgammaClusterAlgos/interface/SCEnergyCorrectorSemiParm.h:150]: This parameter should be added to the `fillPSetDescription` as well.
Why is it set in this function and not in the constructor itself? It is not needed to set any token.

**Comment by swagata87** [comment]
> Hi @ram1123 this is relevant for Run-3 only right? If yes, then I believe it should be under the Run-3 modifier.

Aren't modifiers relevant only for offline reco? 
Ram's PR is specific to HLT supercluster regression.

**Comment by francescobrivio** [comment]
From private conversation with @swagata87 and EGm experts we understood that this change should go together with update of the EGM conditions in the HLT GT.
Note that the official request to update the GT hasn't come yet (@saumyaphor4252 please do it asap!), so we might need a couple of days. I'll bring this up at the ORP tomorrow. FYI @perrotta @qliphy

**Comment by francescobrivio** [comment]
urgent

 - with the caveat described in https://github.com/cms-sw/cmssw/pull/37491#issuecomment-1091946705

File: RecoEcal/EgammaClusterAlgos/src/SCEnergyCorrectorSemiParm.cc [hunk 2]
```diff
@@ -51,6 +51,7 @@ SCEnergyCorrectorSemiParm::SCEnergyCorrectorSemiParm()
       caloGeom_(nullptr),
       isHLT_(false),
       isPhaseII_(false),
+      regTrainedWithPS_(true),
```
Comment (francescobrivio) [inline RecoEcal/EgammaClusterAlgos/src/SCEnergyCorrectorSemiParm.cc:54]: @ram1123 please change this to `false`. As far as I understood this should restore things to the way they are now.
Once we have the new conditions validate we (@cms-sw/alca-l2) we'll take care of changing it to `true` when we update the GT.

**Review by francescobrivio — CHANGES_REQUESTED** [review]
After more discussion with EGM experts and @cms-sw/alca-l2 we came up with a tentative plan:
 - include this PR in the release with `false` flag (since the conditions are not there yet)
 - we cut the 12_3_0 release
 - we (alca) do the FullTrackValidation (where we can change the flag by hand and thus really test the new conditions)
 - we (alca) will do a PR with the new HLT GT and changing the flag to `true`

The only drawback is that 12_3_0 won't have the latest HLT GT, but that can always be added later, and this is a good way to include already in 12_3_0 all the `c++` changes needed.

**Comment by swagata87** [comment]
@ram1123 somewhere in PR description, please make it clear that this PR is relevant ONLY for HLT supercluster regression, and that it does not effect the offline egamma regression.

**Comment by qliphy** [comment]
please test

**Comment by clacaputo** [comment]
assign hlt

**Comment by clacaputo** [comment]
+reconstruction

- PR relevant only for HLT supercluster regression
- boolean set to `false`, see https://github.com/cms-sw/cmssw/pull/37491#pullrequestreview-935418113

**Comment by missirol** [comment]
+hlt

**Comment by perrotta** [comment]
+1
- A new option added, and the default value does not change previous behaviour

**Pull request merged**